pkg/volume/nfs/nfs_test.go

```diff
@@ -136,6 +136,23 @@ func doTestPlugin(t *testing.T, spec *volume.Spec) {
 	if mounter.(*nfsMounter).readOnly {
 		t.Errorf("The volume source should not be read-only and it is.")
 	}
+	mntDevs, err := fake.List()
+	if err != nil {
+		t.Errorf("fakeMounter.List() failed: %v", err)
+	}
+	if len(mntDevs) != 1 {
+		t.Errorf("unexpected number of mounted devices. expected: %v, got %v", 1, len(mntDevs))
+	} else {
+		if mntDevs[0].Type != "nfs" {
+			t.Errorf("unexpected type of mounted devices. expected: %v, got %v", "nfs", mntDevs[0].Type)
+		}
+		src, _, _ := getVolumeSource(spec)
+		srvr := getServerFromSource(src)
+		expectedDevice := fmt.Sprintf("%s:%s", srvr, src.Path)
+		if mntDevs[0].Device != expectedDevice {
+			t.Errorf("unexpected nfs device, expected %q, got: %q", expectedDevice, mntDevs[0].Device)
+		}
```

Comment:
All these checks are good to have, but we do not test anywhere that `[]` was actually added. The test will pass even without it.

There are multiple ways how to solve it, I would probably add a new explicit `expectedDevice` argument to `doTestPlugin()` and did not calculate it inside. But you need to update all the unit tests :-\.